aries_vcx/tests/test_integration.rs

```diff
@@ -1001,38 +1014,104 @@ mod tests {
+        assert_eq!(ConnectionState::Inviter(InviterState::Requested), conn_sender.get_state());
+        conn_sender.update_state().unwrap();
+        assert_eq!(ConnectionState::Inviter(InviterState::Responded), conn_sender.get_state());
 
         consumer.activate().unwrap();
-        consumer_to_institution.update_state().unwrap();
-        assert_eq!(ConnectionState::Invitee(InviteeState::Completed), consumer_to_institution.get_state());
+        thread::sleep(Duration::from_millis(500));
+        conn_receiver.update_state().unwrap();
+        assert_eq!(ConnectionState::Invitee(InviteeState::Completed), conn_receiver.get_state());
 
         institution.activate().unwrap();
         thread::sleep(Duration::from_millis(500));
-        institution_to_consumer.update_state().unwrap();
-        assert_eq!(ConnectionState::Inviter(InviterState::Completed), institution_to_consumer.get_state());
+        conn_sender.update_state().unwrap();
+        assert_eq!(ConnectionState::Inviter(InviterState::Completed), conn_sender.get_state());
 
-        institution_to_consumer.send_generic_message("Hello Alice, Faber here").unwrap();
+        let conn = oob_receiver.connection_exists(vec![&conn_receiver]).unwrap();
+        assert!(conn.is_some());
+
+        let a2a_msg = oob_receiver.extract_a2a_message().unwrap().unwrap();
+        assert!(matches!(a2a_msg, A2AMessage::PresentationRequest(..)));
+        if let A2AMessage::PresentationRequest(request_receiver) = a2a_msg {
+            assert_eq!(request_receiver.thread.unwrap().pthid.unwrap(), oob_receiver.id.0);
+            assert_eq!(request_receiver.request_presentations_attach, request_sender.request_presentations_attach);
+        }
+    }
+
+    #[test]
+    #[cfg(feature = "agency_pool_tests")]
+    fn test_connection_reuse() {
+        let _setup = SetupLibraryAgencyV2::init();
+        let mut institution = Faber::setup();
+        let mut consumer = Alice::setup();
+
+        let (consumer_to_institution, institution_to_consumer) = create_connected_connections_via_public_invite(&mut consumer, &mut institution);
+
+        institution.activate().unwrap();
+        let service = FullService::try_from(&institution.agent).unwrap();
+        let oob_sender = OutOfBand::create()
+            .set_label("test-label")
+            .set_goal_code(GoalCode::P2PMessaging)
+            .set_goal("To exchange message")
+            .append_service(ServiceResolvable::FullService(service));
+        let oob_msg = oob_sender.to_a2a_message();
 
         consumer.activate().unwrap();
-        let consumer_msgs = consumer_to_institution.download_messages(Some(vec![MessageStatusCode::Received]), None).unwrap();
-        assert_eq!(consumer_msgs.len(), 1);
+        let oob_receiver = OutOfBand::create_from_a2a_msg(&oob_msg).unwrap();
+        let conn = oob_receiver.connection_exists(vec![&consumer_to_institution]).unwrap();
+        assert!(conn.is_some());
```

Comment:
ideally let's have Alice create more than 1 connection and make sure that `connection_exists` picks the right one. 
We should also test it returns None, if no matching one is supplied

Reply:
We could also test the connection will not match if the original connection was created "privately", instead of  public did invite.

Reply:
That's right. Ideally, I think that testing the functionality of `connection_exists` specifically should be a job of a couple unit tests.